nano/node/repcrawler.cpp

```diff
@@ -324,6 +324,10 @@ void nano::rep_crawler::cleanup_reps ()
 				equal = true;
 			}
 		}
+		else if (i->get_type () == nano::transport::transport_type::fake)
```

Comment:
Is this still needed? I think it might not be needed and we do not want to have code that checks the type of an object, it breaks polymorphism.

Reply:
Unfortunately it it is needed. If this line is removed then `cleanup_reps()` will always remove the rep with the fake channel and this is a race condition in the test. It could lead to an assertion failure in line 1859 in core_test/node.cpp:
```
ASSERT_FALSE (searching_node.rep_crawler.response (channel_rep1, vote_rep1, true));
ASSERT_TIMELY (5s, searching_node.rep_crawler.representative_count () == 1);
```
The second assert could fail because the rep could be removed immediately after `rep_crawler.response(...)` was called. So that's why this line is needed.